File: Chapter6/R/Example5.R
# Chapter 6 - Κανονική Κατανομή
# Example 5: Σύγκριση Κατανομών
#   N(50, 5²), N(50, 10²), N(60, 5²)

# x range
x <- seq(20, 80, length.out = 1000)

# Three distributions
y1 <- dnorm(x, mean = 50, sd = 5)
y2 <- dnorm(x, mean = 50, sd = 10)
y3 <- dnorm(x, mean = 60, sd = 5)

# Plot
plot(x, y1, type = 'l', lwd = 2, col = 'blue',
     xlab = 'x', ylab = 'f(x)',
     main = 'Comparison of Normal Distributions',
     ylim = c(0, max(c(y1, y2, y3))))

lines(x, y2, lwd = 2, col = 'red')
lines(x, y3, lwd = 2, col = 'green')

legend('topright',
       legend = c('N(50, 5^2)', 'N(50, 10^2)', 'N(60, 5^2)'),
       col    = c('blue', 'red', 'green'),
       lwd    = 2)

grid()

File: Chapter6/Python/Example5.py
# Chapter 6 - Κανονική Κατανομή
# Example 5: Σύγκριση Κατανομών
#   N(50, 5²), N(50, 10²), N(60, 5²)

import numpy as np
import matplotlib.pyplot as plt
from scipy.stats import norm

# x range
x = np.linspace(20, 80, 1000)

# Three distributions
y1 = norm.pdf(x, loc=50, scale=5)
y2 = norm.pdf(x, loc=50, scale=10)
y3 = norm.pdf(x, loc=60, scale=5)

# Plot
plt.figure(figsize=(12, 7))
plt.plot(x, y1, 'b-', linewidth=2, label='N(50, 5\u00b2)')
plt.plot(x, y2, 'r-', linewidth=2, label='N(50, 10\u00b2)')
plt.plot(x, y3, 'g-', linewidth=2, label='N(60, 5\u00b2)')

plt.xlabel('x', fontsize=12)
plt.ylabel('f(x)', fontsize=12)
plt.title('Comparison of Normal Distributions', fontsize=14)
plt.legend(fontsize=11)
plt.grid(alpha=0.3)
plt.tight_layout()
plt.show()
